### PR TITLE
[RyuJIT] Delete dead code

### PR DESCRIPTION
**Pull request by mikedn**
optFindLocalInit and related functions (optIsTrackedLocal, lvaLclVarRefs, lvaLclVarRefsAccumIntoRes, lvaLclVarRefsAccum) are not used anywhere.
Also delete a bunch of undefined function declarations.

### CODE REVIEW

**Comment by BruceForstall** [comment]
@dotnet/jit-contrib

**Comment by BruceForstall** [comment]
@dotnet-bot test this please

**Comment by sandreenko** [comment]
LGTM, looks like this code was replaced by SSA based algorithm that is located in `earlyprop.cpp` now. That happened in 2015.

File: src/jit/dataflow.h [hunk 1]
```diff
@@ -24,20 +24,16 @@ class DataFlow
...
-        Callback(Compiler* pCompiler);
-
-        void StartMerge(BasicBlock* block);
-        void Merge(BasicBlock* block, BasicBlock* pred, flowList* preds);
-        bool EndMerge(BasicBlock* block);
-
-    private:
-        Compiler* m_pCompiler;
-    };
+    // The callback class that needs to be implemented by DataFlow users
```
[the remark this answers is not in the document]
Comment (mikedn) [reply src/jit/dataflow.h]: Thanks, I wasn't very happy with "users" but couldn't think of something better at the time.

On a closer look, the whole comment is also misplaced, it has nothing to do with the class constructor and there's another comment at the top of the file that also mentions the callback, it makes more sense to just move the callback example there.

**Comment by mikedn** [comment]
On a side note - the dead function declarations were found using VS, its editor highlights them in green. However, it gets confusing by things like ifdefs and macros so I double checked by searching all JIT sources that they are indeed not defined anywhere.

**Comment by sandreenko** [comment]
@dotnet-bot test Windows_NT arm Cross Checked Innerloop Build and Test

**Comment by BruceForstall** [comment]
Lots of bad luck with the CI... Trying again...

@dotnet-bot test Ubuntu arm Cross Checked Innerloop Build and Test
@dotnet-bot test Ubuntu arm Cross Checked no_tiered_compilation_innerloop Build and Test
@dotnet-bot test Windows_NT arm64 Cross Checked Innerloop Build and Test

**Comment by mikedn** [comment]
Yay, CI is green, merge it while it doesn't change its mind :)

**Pull request merged**